plenum/bls/bls_bft_replica_plenum.py

```diff
@@ -157,17 +157,12 @@ def validate_key_proof_of_possession(self, key_proof, pk):
             .verify_key_proof_of_possession(key_proof, pk)
 
     def _validate_signature(self, sender, bls_sig, pre_prepare: PrePrepare):
-        if f.POOL_STATE_ROOT_HASH.nm in pre_prepare:
-            pool_root_hash = self.state_root_serializer.deserialize(pre_prepare.poolStateRootHash)
-            pool_root_hash_ser = pre_prepare.poolStateRootHash
-        else:
-            pool_root_hash = self._bls_bft.bls_key_register.get_pool_root_hash_committed()
-            pool_root_hash_ser = self.state_root_serializer.serialize(bytes(pool_root_hash))
-
+        pool_root_hash = self._get_pool_root_hash(pre_prepare)
         sender_node = self.get_node_name(sender)
         pk = self._bls_bft.bls_key_register.get_key_by_name(sender_node, pool_root_hash)
         if not pk:
             return False
+        pool_root_hash_ser = self._get_pool_root_hash(pre_prepare, serialize=False)
```

Comment:
I think serialized value needs to be passed here (serialize=True).
In general, I would suggest to move calculation of pool_root_hash_ser to `_create_multi_sig_value_for_pre_prepare` method.

Reply:
Fixed